Configuration/UTMAppleConfigurationDrive.swift

```diff
@@ -92,12 +97,18 @@ struct UTMAppleConfigurationDrive: UTMConfigurationDrive {
             try container.encodeIfPresent(imageName, forKey: .imageName)
         }
         try container.encode(isReadOnly, forKey: .isReadOnly)
+        try container.encode(isNvme, forKey: .isNvme)
         try container.encode(id, forKey: .identifier)
     }
     
-    func vzDiskImage() throws -> VZDiskImageStorageDeviceAttachment? {
+    func vzDiskImage(useFsWorkAround: Bool = false) throws -> VZDiskImageStorageDeviceAttachment? {
         if let imageURL = imageURL {
-            return try VZDiskImageStorageDeviceAttachment(url: imageURL, readOnly: isReadOnly)
+            // Use cached caching mode for virtio drive to prevent fs corruption on linux when possible
+            if #available(macOS 12.0, *), !isNvme, useFsWorkAround {
+                return try VZDiskImageStorageDeviceAttachment(url: imageURL, readOnly: isReadOnly, cachingMode: .cached, synchronizationMode: .fsync)
```

Comment:
In my case, `synchronizationMode` defaults to `.full` when `cachingMode` and `synchronizationMode` aren't supplied.
```
print(mainDiskAttachment.synchronizationMode.rawValue) // -> 1 == .full
// https://developer.apple.com/documentation/virtualization/vzdiskimagesynchronizationmode/full
```
Does it default to `.fsync` for you? I didn't notice any differences in terms of corruption for any synchronization modes, so I'd say we should stick to framework default.

Reply:
I'm using `.fsync` because it could provide better performance in theory. I did not notice that the framework defaults to `.full`.

Reply:
Changed to `.full` now.